test/basic.test.js

```diff
@@ -147,6 +147,30 @@ test('boot a plugin with options', (t) => {
   })
 })
 
+test('boot a plugin with a function that returns the options', (t) => {
+  t.plan(4)
+
+  const server = {}
+  const app = boot(server)
+  const myOpts = {
+    hello: 'world'
+  }
+  const myOptsAsFunc = parent => {
+    t.strictEqual(parent, app)
+    return myOpts
+  }
+
+  app.use(function (s, opts, done) {
+    t.equal(s, server, 'the first argument is the server')
+    t.deepEqual(opts, myOpts, 'passed options via function')
+    done()
+  }, myOptsAsFunc)
+
+  app.on('start', () => {
+    t.pass('booted')
+  })
+})
```

Comment:
Can you also add a test that demonstrates this sentence:

> This is useful in cases where an injected variable from a plugin needs to be made available to another.

It's the whole point of this change, and we need to verify that we don't regress.

Reply:
Thank you for taking your time to vet the code carefully. The fix you had purposed for passing in the parent variable to the context of the function failed as it was evaluated at the time the `Plugin` was created. I fixed it by utilizing `bind` and executing it during `Plugin.prototype.exec` which fixed the issue.